api/v1beta2/awsmachine_types.go

```diff
@@ -153,6 +163,11 @@ type AWSMachineSpec struct {
 	// +kubebuilder:validation:MaxItems=2
 	NetworkInterfaces []string `json:"networkInterfaces,omitempty"`
 
+	// NetworkInterfaceType is the interface type of the primary network Interface.
+	// +kubebuilder:validation:Enum=interface;efa
+	// +optional
+	NetworkInterfaceType *NetworkInterfaceType `json:"networkInterfaceType,omitempty"`
```

Comment:
Does this need to be a pointer? There's no difference between `""` and a nil value right?

Reply:
I used the pointer to avoid "" value being passed through to the AWS API. 

Changed to non-pointer and updated the code to not set empty string on the AWS API call.

Reply:
I might be missing something, but isn't `""` passed to the AWS API due to Go's defaulting to zero values?

Reply:
The AWS API call still uses *string. I added a condition to keep the value nil when "" value is on the spec. https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/5275/files#diff-4d2ea189a70ea097ffeeb50f78cc676a224386d4fe17b1bec9c83a4731d9c379R586-R589